service/host.go

```diff
@@ -247,19 +270,23 @@ func (s *host) Start(waitForShutdown bool) (<-chan Exit, <-chan struct{}, error)
 				if _, err := s.shutdown(e, "", nil); err != nil {
 					s.Logger().Error("Unable to shut down modules", "initialError", e, "shutdownError", err)
 				}
-				return errChan, readyCh, e
+				return Control{
+					ExitChan:     errChan,
+					ReadyChan:    readyCh,
+					ServiceError: e,
```

Comment:
Current behavior seems to be the correct behavior to me. Returning the first error you get is probably essential for debugging rather than knowing all errors a, b, c...